File: .skills/openclaw-skills/skills/orlyjamie/not-nasty-skill/greet.ts
/**
 * Simple greeting utility
 */

export function greet(name: string): string {
  const greetings = [
    `Hello, ${name}!`,
    `Hey there, ${name}!`,
    `Hi ${name}, nice to see you!`,
    `Greetings, ${name}!`,
  ];

  const randomIndex = Math.floor(Math.random() * greetings.length);
  return greetings[randomIndex];
}

export function getTimeBasedGreeting(name: string): string {
  const hour = new Date().getHours();

  if (hour < 12) {
    return `Good morning, ${name}!`;
  } else if (hour < 17) {
    return `Good afternoon, ${name}!`;
  } else {
    return `Good evening, ${name}!`;
  }
}

File: .skills/openclaw-skills/skills/chndranndr/japanese-tutor/scripts/greet.py
import datetime
import random

def get_greeting():
    hour = datetime.datetime.now().hour
    
    if 5 <= hour < 11:
        greetings = ["Ohayou Gozaimasu (Good Morning) ☀️", "Ohayou!"]
    elif 11 <= hour < 18:
        greetings = ["Konnichiwa (Hello) 🌤️", "Yahho!"]
    else:
        greetings = ["Konbanwa (Good Evening) 🌙", "Otsukare!"]
        
    return random.choice(greetings)

if __name__ == "__main__":
    print(get_greeting())
